### PR TITLE
Allow groups to restrict by browser integration key (#6437)

### PR DESCRIPTION
**Pull request by ekimd**
Relates to issue #6437

This branch adds the ability to set a browser restriction on groups.  This allows a given group tree to be visible only to a specific browser.  Previously, the workaround was to have separate databases for each browser.  However, this can become cumbersome under situations where someone may have a dozen or more browsers that each have to be unlocked whenever the screensaver activates.

## Screenshots
image

## Testing strategy
- Added a new test function to test the new behavior

## Type of change
- ✅ New feature (change that adds functionality)

### CODE REVIEW

**Comment by ekimd** [comment]
@droidmonkey I don't understand the codecov/patch failure--what do I need to do to fix that?
Thanks

**Comment by droidmonkey** [comment]
We set an ideal threshold for new code introduced to be 75% covered by unit tests. It's optional though.

**Comment by ekimd** [comment]
Thanks.  I tried increasing the test coverage but couldn't find a clear path given that some of the parent functions in BrowserService aren't tested.

How do I request a code review?

**Comment by varjolintu** [comment]
Could you solve the conflicts? Rebase the branch.

**Comment by ekimd** [comment]
Rebase complete

**Comment by varjolintu** [comment]
Some of the code could be cleaned a little bit. For example:
- Use `QStringList` instead if `QList<QString>` if possible.
- Use `auto` instead of `QString` when creating a new variable.
- Return `{}` instead of `QString()`.
- If function parameter has a default value, use `{}` again.

**Comment by ekimd** [comment]
Thanks for the fast response!

I think I caught them all, but let me know if there's anything else.  I couldn't return {} in BroswerService.cpp:996 because I got a compiler error returning an initializer list from that lambda function.

File: src/browser/BrowserService.cpp
Comment (varjolintu) [inline src/browser/BrowserService.cpp]: Let's keep this consistent with other code:
```
if (key.isEmpty()) {
    return tr("Disable");
} else {
    return key;
}
```

Or just:
`return key.isEmpty() ? tr("Disable") : key;`

**Comment by varjolintu** [comment]
Tested this, and it works nicely. Only thing that caught my eye is the strange separator in the combobox:
Screenshot 2023-11-01 at 7 01 48

When looking at other comboboxes, the separators look much more smaller:
Screenshot 2023-11-01 at 7 01 30

**Comment by ekimd** [comment]
Interesting.  I think that's because the screenshot above is actually referencing a QMenu not a ComboBox (see EditWidgetIcons.cpp:170).

**Review by varjolintu — APPROVED** [review]
Approved. We are missing a combobox separator style, but the scope is outside of this PR.

**Comment by ekimd** [comment]
Anything else I need to do to get approval on this?

**Comment by ekimd** [comment]
bump

**Comment by varjolintu** [comment]
> bump

Bump? This PR is already accepted.

**Comment by ekimd** [comment]
Going to need some help here.  There was a merge conflict so I resolved that but now one of the unit tests is failing and I'm not sure why.  (It seems unrelated)

**Comment by droidmonkey** [comment]
You can ignore, we have sporadic failures since some tests rely on randomness and infrequently don't produce the expected results.

**Comment by ekimd** [comment]
Since I'm not familiar with the development cadence for this project, what is the typical timeline/workflow between approval, review, and merge?

**Comment by droidmonkey** [comment]
I will merge it in when I get the chance. Unfortunately my time is very limited lately.

**Comment by yoris** [comment]
This option is very useful but it wasn't included in 2.7.11 as well.

**Comment by ekimd** [comment]
I'm using the Fedora stock 2.7.11 build and the "Restrict matching to given
browser key" is present and functional.  (See attached screenshot)[image:
image.png]

**Comment by yoris** [comment]
All looks good—the option exists. Apologies—I was searching in the incorrect section of the UI settings.

**Pull request merged**
